lib/tests/select.rs

```diff
@@ -107,3 +107,84 @@ async fn select_writeable_subqueries() -> Result<(), Error> {
 	//
 	Ok(())
 }
+
+#[tokio::test]
+async fn select_where_field_is_bool() -> Result<(), Error> {
+	let sql = "
+		CREATE test:1 SET active = false;
+		CREATE test:2 SET active = false;
+		CREATE test:3 SET active = true;
+		SELECT * FROM test WHERE active = false;
+		SELECT * FROM test WHERE active != true;
+	";
+
+	let dbs = Datastore::new("memory").await?;
+	let ses = Session::for_kv().with_ns("test").with_db("test");
+	let res = &mut dbs.execute(&sql, &ses, None, false).await?;
+	assert_eq!(res.len(), 5);
+	//
+	let tmp = res.remove(0).result?;
```

Comment:
@tobiemh I know we typically run unit tests against the `Datastore` directly. One of the reasons for this being that most of our unit tests precede the API layer but using `remove(0)` multiple times feels kind of fragile to me. If you delete one of those, it throws off the rest of the following tests. It also makes it harder to follow which statement we are testing against when reading the code.

If we use `Surreal::query` instead, we can do `take(0)`, `take(1)`, `take(2)` etc. This makes the code easier to read and it's not dependent on order of removal so it will work regardless of the order you remove statement results which means the tests will still work even if you delete an earlier block. So what do you think about using `Surreal::query` here?

Reply:
I completely agree, especially when it comes to readability.

> I'm going to go ahead and approve this PR for now because it follows the status quo. If we decide to follow my proposal on dropping `remove(0)` we can still do it in this PR or a future PR.

I think it's more appropriate to create another PR. The other tests in that same file also use `remove(0)`. That way, we can refactor all of them together if it comes to it.

I'll go ahead and mark this as resolved, so we can merge this.

Reply:
@rushmorem if you were to remove a line/statement from the test query, would you have to update all of the `take(usize)` method calls?

Reply:
Yes @tobiemh you would need to update all those that previously had an index bigger than the index of the statement you removed. The number in `take(usize)` corresponds to the index of each statement in the query.